src/Chessboard.cpp

```diff
@@ -0,0 +1,313 @@
+#include "Chessboard.h"
+#include "Rook.h"
+#include "Bishop.h"
+#include "King.h"
+#include "Queen.h"
+#include "Knight.h"
+#include "Pawn.h"
+
+Chessboard::Chessboard() {
+    initializeBoard();
+    currentPlayer_ = PieceColor::White;
+}
+
+
+Chessboard::~Chessboard() {
+    clearBoard();
+}
+
+void Chessboard::initializeBoard() {
+    for (int row = 0; row < 8; row++) {
+        for (int col = 0; col < 8; col++) {
+            board_[row][col] = nullptr;
+        }
+    }
+
+    // initialize white pieces
+    board_[0][0] = new Rook(PieceColor::White, 0, 0);
+    board_[0][1] = new Knight(PieceColor::White, 0, 1);
+    board_[0][2] = new Bishop(PieceColor::White, 0, 2);
+    board_[0][3] = new Queen(PieceColor::White, 0, 3);
+    board_[0][4] = new King(PieceColor::White, 0, 4);
+    board_[0][5] = new Bishop(PieceColor::White, 0, 5);
+    board_[0][6] = new Knight(PieceColor::White, 0, 6);
+    board_[0][7] = new Rook(PieceColor::White, 0, 7);
+
+    for (int i = 0; i < 8; i++) {
+        board_[1][i] = new Pawn(PieceColor::White, 1, i);
+    }
+
+    // initialize black pieces
+    board_[7][0] = new Rook(PieceColor::Black, 7, 0);
+    board_[7][1] = new Knight(PieceColor::Black, 7, 1);
+    board_[7][2] = new Bishop(PieceColor::Black, 7, 2);
+    board_[7][3] = new Queen(PieceColor::Black, 7, 3);
+    board_[7][4] = new King(PieceColor::Black, 7, 4);
+    board_[7][5] = new Bishop(PieceColor::Black, 7, 5);
+    board_[7][6] = new Knight(PieceColor::Black, 7, 6);
+    board_[7][7] = new Rook(PieceColor::Black, 7, 7);

```

Comment:
can be simplified and made more readable by breaking it down into smaller functions with clear responsibilities. This can improve code maintainability and readability.